### PR TITLE
Add support for ppc64, arm64, s390x

### PR DESCRIPTION
**Pull request by adam-stokes**
Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

### CODE REVIEW

**Review by tvansteenburgh — APPROVED** [review]
LGTM, thanks.

File: snap/Dockerfile.aarch64 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM arm64v8/ubuntu:xenial
```
Comment (hyperbolic2346) [inline snap/Dockerfile.aarch64:1]: Should this file be named Dockerfile.arm64?

File: snap/Dockerfile.aarch64 [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+RUN apt-get update && \
+  apt-get dist-upgrade --yes && \
+  apt-get install --yes \
+  snapcraft curl file sudo jq squashfs-tools && \
```
Comment (Cynerva) [inline snap/Dockerfile.aarch64]: This one doesn't install `make`, but the others do. Is that intentional?

File: snap/Dockerfile.aarch64 (hunk 1 above)
Comment (hyperbolic2346) [reply snap/Dockerfile.aarch64:1]: oh no, does `uname -m` return aarch64 for arm64?

File: snap/build-scripts/docker-build [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 
-docker build -t docker-build .
+set -eux
+
+docker build -f "./Dockerfile.$(uname -m)" -t docker-build .
```
Comment (hyperbolic2346) [inline snap/build-scripts/docker-build:5]: So every time we build we're going to update the build tool containers? We don't want this as a different step so we can update them, but not force it on every build?

File: snap/Dockerfile.aarch64 (hunk 2 above)
Comment (adam-stokes) [reply snap/Dockerfile.aarch64]: oversigh

File: snap/build-scripts/docker-build (hunk 3 above)
Comment (adam-stokes) [reply snap/build-scripts/docker-build:5]: im down for not refreshing new containers, but i suck at docker so feel free to tack on to this

File: snap/Dockerfile.aarch64 (hunk 1 above)
Comment (adam-stokes) [reply snap/Dockerfile.aarch64:1]: yea :\

**Pull request merged**